### DISCUSSION TITLE
Is there a memory limitation?

### DISCUSSION THREAD

**Discussion by shi-yan** [tauri-apps/tauri#6461] [Q&A]
I'm wondering if tauri has the same memory limitation of a browser tab (4gb I remember?)

I'm trying to work on a data-dense app. I'm wondering if I need to worry about memory.

Thanks

**Comment by vdvman1** [comment]
You could store the bulk of the data in the Rust backend of the app, and only keep the data you need at any given time in the webview frontend

**Reply by FabianLars** [reply]
^ p.s. yes, the same memory limitations of a browser apply to the webview process (your js) in tauri, the core/backend (rust) has no limitation.

**Reply by shi-yan** [reply]
how does the js <-> rust communication work? is there latency?

**Reply by FabianLars** [reply]
yes there is. pretty neglectable for small infrequent transfer, but if you need to send huge amount of payloads the current IPC will really start to choke on it because everything you send over it will have to be converted to a json string.
in v2 we're trying to solve this by using different webview apis that do not force us to use strings...
